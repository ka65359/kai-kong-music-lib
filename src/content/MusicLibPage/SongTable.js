import React from "react";
import {
  DataTable,
  TableContainer,
  Table,
  TableHead,
  TableRow,
  TableHeader,
  TableBody,
  TableCell
} from "carbon-components-react";
import "./_song-table.scss";

const SongTable = ({ rows, headers }) => {
  return (
    <DataTable
      rows={rows}
      headers={headers}
      render={({
        rows,
        headers,
        getHeaderProps,
        getRowProps,
        getTableProps
      }) => (
        <TableContainer
          title="My Music Library"
          description="A collection of music.">
          <Table {...getTableProps()}>
            <TableHead>
              <TableRow>
                {headers.map((header) => (
                  <TableHeader
                    key={header.header}
                    {...getHeaderProps({ header })}>
                    {header.header}
                  </TableHeader>
                ))}
              </TableRow>
            </TableHead>
            <TableBody>
              {rows.map((row) => (
                <React.Fragment key={row.id}>
                  <TableRow {...getRowProps({ row })}>
                    {row.cells.map((cell) => (
                      <TableCell key={cell.id}>{cell.value}</TableCell>
                    ))}
                  </TableRow>
                </React.Fragment>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      )}
    />
  );
};

export default SongTable;
