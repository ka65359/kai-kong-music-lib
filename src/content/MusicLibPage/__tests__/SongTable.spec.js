import React from "react";
import { shallow } from "enzyme";
import SongTable from "../SongTable";

describe("SongTable() tests", () => {
  it("should render correctly in mode", () => {
    const component = shallow(<SongTable />);

    expect(component).toMatchSnapshot();
  });
});
