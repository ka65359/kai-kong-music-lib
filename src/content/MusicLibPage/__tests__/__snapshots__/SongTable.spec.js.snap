// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongTable() tests should render correctly in mode 1`] = `ShallowWrapper {}`;
